robot_controllers/src/parallel_gripper.cpp

```diff
@@ -57,10 +57,51 @@ int ParallelGripperController::init(ros::NodeHandle& nh, ControllerManager* mana
 
   // Setup Joints */
   std::string l_name, r_name;
-  nh.param<std::string>("l_gripper_joint", l_name, "l_gripper_finger_joint");
-  nh.param<std::string>("r_gripper_joint", r_name, "r_gripper_finger_joint");
-  left_ = manager_->getJointHandle(l_name);
-  right_ = manager_->getJointHandle(r_name);
+
+//Checking to see if the joints have been set in the parameter server
+//Warns and attempts to use default joint names
+
+  if(!nh.hasParam("l_gripper_joint"))
+    {  
+     ROS_ERROR_NAMED("ParallelGripperController",
+                    "Gripper joint parameter 'l_gripper_joint' has not been set");
+     ROS_WARN_NAMED("ParallelGripperController",
+                    "Gripper Controller:Attempting to start using default joint names,may not work with simulation");
+   
+     nh.param<std::string>("l_gripper_joint", l_name, "l_gripper_finger_joint");
+     left_ = manager_->getJointHandle(l_name);
+    }
+    
+  if(!nh.hasParam("r_gripper_joint"))
+    {
+    ROS_ERROR_NAMED("ParallelGripperController",
+                    "Gripper joint parameter 'r_gripper_joint' has not been set");
+    ROS_WARN_NAMED("ParallelGripperController",
+                    "Gripper Controller:Attempting to start using default joint names,may not work with simulation");
+    nh.param<std::string>("r_gripper_joint", r_name, "r_gripper_finger_joint");
+    right_ = manager_->getJointHandle(r_name);
+    }
+
+// This prevents it from causing a segmentation fault
+ 
+
+  if(!left_)
+    {
+    ROS_ERROR_NAMED("ParallelGripperController",
```

Comment:
I think this message should have more information, like what joint name parameter value was used and what ROS parameter (in what namespace) to change.  Something like:
```
ROS_ERROR_NAMED("ParallelGripperController", "Unable to find joint name '%s' for left finger joint."
                                         "To modify value,  change parameter '%s/l_gripper_joint'",  
                                         l_name.c_str(), nh.getNamespace().c_str())
```

In the error message I've included the namespace because it might not be clear because some controllers might be used multiple times (like for a gripper controller for a robot with two arms).